src/SDKs/Azure.Core/data-plane/Azure.Core/Pipeline/Policies/RetryPolicy.cs

```diff
@@ -93,7 +97,30 @@ internal virtual async Task DelayAsync(TimeSpan time, CancellationToken cancella
             await Task.Delay(time, cancellationToken).ConfigureAwait(false);
         }
 
-        protected abstract void GetDelay(HttpPipelineMessage message, int attempted, out TimeSpan delay);
+        protected virtual void GetDelay(HttpPipelineMessage message, int attempted, out TimeSpan delay)
+        {
+            delay = TimeSpan.Zero;
+            if (message.Response.TryGetHeader(RetryAfterHeaderName, out var retryAfterValue))
```

Comment:
I wonder if this should not be a setting on retyr policy, i.e. enabling disabling honoring the header and which one. I don't mind the default for this setting to be On and try all headers, butr I think we might want to have an option to disable/configure this. This would allow client libraries to not worry about the order in which the headers are inspected

Reply:
Fells like too much customization. One thing we can do it to turn `GetDelay` into a separate virtual method e.g. `GetHeaderBaseDelay` so if clients want they can override its implementation without affecting other functionality.

Reply:
Let's just extract the logic into a separate vitrual method